feat: new .NET plugin (v2) (#1047)
canonical/craft-parts: docs/common/craft-parts/reference/plugins/dotnet_plugin.rst

The ``dotnet`` plugin builds .NET projects using the ``dotnet`` tool.

> .. warning::

Author: Should I remove both that warning and the [note](https://github.com/mateusrodrigues/craft-parts/blob/edc951f6e6ec8bf1069ac430b1ec4d920adfeeb1/docs/common/craft-parts/reference/plugins/dotnet_v2_plugin.rst?plain=1#L8-L10) in the v2 documentation then?

> .. warning::

Author: Ok, I updated the notes accordingly.

The ``dotnet`` plugin builds .NET projects using the ``dotnet`` tool.

> .. warning::
>     This plugin has been superseded by :ref:`.NET plugin (v2) <craft_parts_dotnet_v2_plugin>`.

Reviewer: ```suggestion
.. admonition:: Legacy plugin

    This plugin is superseded by the :ref:`craft_parts_dotnet_v2_plugin`.
```
